VEdger/Repository/EntityRepository.cs

```diff
@@ -1,38 +1,66 @@
-﻿using System;
-using System.Collections.Generic;
-
-namespace VEdger.Repository
+﻿namespace VEdger.Repository
 {
-    public class EntityRepository<T> : IRepository<T>
+    using System;
+    using System.Collections.Generic;
+    using System.Data.Entity;
+    using System.Linq;
+    using VEdger.Models;
+
+    public class EntityRepository<TEntity> : IRepository<TEntity> where TEntity : class
     {
-        public void Create<T>(T entity)
+        private readonly ApplicationDbContext context;
+        private readonly DbSet<TEntity> dbSet;
+
+        public EntityRepository()
         {
-            throw new NotImplementedException();
+            this.context = new ApplicationDbContext();
+            this.dbSet = context.Set<TEntity>();
         }
 
-        public void Delete<T>(T entityForDeletion)
+        public void Create<T>(
+            TEntity entity)
         {
-            throw new NotImplementedException();
+            this.dbSet.Add(entity);
+            context.SaveChangesAsync();
         }
 
-        public IEnumerable<T> Read<T>()
+        public void Delete<T>(
+            TEntity entityForDeletion)
         {
             throw new NotImplementedException();
         }
 
+        public IEnumerable<TEntity> Read<TEntity>()
+        {
+            var data = this.dbSet.ToList();
+            return (IEnumerable<TEntity>)data;
+        }
+
         public T Read<T>(int id)
         {
             throw new NotImplementedException();
         }
 
-        public IEnumerable<T> Read<T>(T parameters)
+        public IEnumerable<T> Read<T>(
+            T parameters)
         {
             throw new NotImplementedException();
         }
```

Comment:
czy te notimplementet i ogolnie definicje metod generycznych sa potrzebne?

Reply:
Racja, jedna z tych metod jest niepotrzeba

Reply:
Not implementet  jest potrzebne,  potem sie napisze w te miejsce implementacje

Reply:
Metody generyczne są potrzebne bo teraz możemy używać jednej klasy  "Entity Repository" dla wszystkich modeli i nie musimy powtarzać kodu